partitioned-table.md

```diff
@@ -164,6 +164,115 @@ Range partitioning is particularly useful when one or more of the following cond
+    job_code int,
+    store_id int NOT NULL
+) PARTITION BY RANGE (id)
+INTERVAL (100) FIRST PARTITION LESS THAN (100) LAST PARTITION LESS THAN (10000) MAXVALUE PARTITION
+```
+
+Which creates the following table:
+
+```
+CREATE TABLE `employees` (
+  `id` int unsigned NOT NULL,
+  `fname` varchar(30) DEFAULT NULL,
+  `lname` varchar(30) DEFAULT NULL,
+  `hired` date NOT NULL DEFAULT '1970-01-01',
+  `separated` date DEFAULT '9999-12-31',
+  `job_code` int DEFAULT NULL,
+  `store_id` int NOT NULL
+)
+PARTITION BY RANGE (`id`)
+(PARTITION `P_LT_100` VALUES LESS THAN (100),
+ PARTITION `P_LT_200` VALUES LESS THAN (200),
+...
+ PARTITION `P_LT_9900` VALUES LESS THAN (9900),
+ PARTITION `P_LT_10000` VALUES LESS THAN (10000),
+ PARTITION `P_MAXVALUE` VALUES LESS THAN (MAXVALUE))
+```
+
+It also works with RANGE COLUMNS partitioning:
+
+```
+CREATE TABLE monthly_report_status (
+    report_id int NOT NULL,
+    report_status varchar(20) NOT NULL,
+    report_date date NOT NULL
+)
+PARTITION BY RANGE COLUMNS (report_date)
+INTERVAL (1 MONTH) FIRST PARTITION LESS THAN ('2000-01-01') LAST PARTITION LESS THAN ('2025-01-01')
+```
+
+Which creates this table:
+
+```
+CREATE TABLE `monthly_report_status` (
+  `report_id` int(11) NOT NULL,
+  `report_status` varchar(20) NOT NULL,
+  `report_date` date NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+PARTITION BY RANGE COLUMNS(`report_date`)
+(PARTITION `P_LT_2000-01-01` VALUES LESS THAN ('2000-01-01'),
+ PARTITION `P_LT_2000-02-01` VALUES LESS THAN ('2000-02-01'),
+...
+ PARTITION `P_LT_2024-11-01` VALUES LESS THAN ('2024-11-01'),
+ PARTITION `P_LT_2024-12-01` VALUES LESS THAN ('2024-12-01'),
+ PARTITION `P_LT_2025-01-01` VALUES LESS THAN ('2025-01-01'))
+```
+
+The optional NULL PARTITION would create a partition where the partitioning expression evaluates to NULL would be placed. In the partitioning expression, NULL is considered to be less than any other value, see [Handling of NULL with Range partitioning](#handling-of-null-with-range-partitioning).
+
+The optional MAXVALUE PARTITION would create a last partition as `PARTITION P_MAXVALUE VALUES LESS THAN (MAXVALUE)`
+
+#### ALTER INTERVAL Partitioned tables
+
+INTERVAL partitioning also adds simpler syntax for adding and dropping partitions.
+
+Changing the first partition, meaning dropping partitions with lower ranges/older data.
+
+```
+ALTER TABLE table_name FIRST PARTITION LESS THAN (<expression>)
+```
+
+Will drop all partitions whose value is lower than the given expression, making the matched partition the new first partition. (It will not affect a NULL PARTITION)
+
+Changing the last partition, meaning adding more partitions with higher ranges/room for new data.
+
+```
+ALTER TABLE table_name LAST PARTITION LESS THAN (<expression>)
+```
+
+Will add new partitions with the current INTERVAL up to and including the given expression. (It will not work if a MAXVALUE PARTITION exists, since that would need data reorganisation).
```

Comment:
```suggestion
It will add new partitions with the current INTERVAL up to and including the given expression. It does not work if a `MAXVALUE PARTITION` exists, because it needs data reorganisation.
```